assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -55,18 +55,19 @@ const RouteVehicles = ({
   const vehicles = useVehiclesForRoute(socket, selectedVehicleRoute)
   return (
     <>
-      {filterVehicles(vehicles).map((vehicle: Vehicle) => {
-        const isSelected = vehicle.id === selectedVehicleId
-        return (
-          <VehicleMarker
-            key={vehicle.id}
-            vehicle={vehicle}
-            isPrimary={isSelected === true}
-            isSelected={isSelected}
-            onSelect={onPrimaryVehicleSelect}
-          />
-        )
-      })}
+      {filterVehicles(vehicles)
+        .filter((vehicle) => vehicle.id !== selectedVehicleId)
```

Comment:
is it possible that the selected vehicle from `useVehiclesForRoute` has newer data than the vehicle returned from `useVehicleById`?

Reply:
It's possible that one will be ever so slightly newer than the other, but basically the way it works on the backend is that once new vehicle data comes in and is processed, it pushes the update out on all applicable channels with vehicle subscriptions, so while there may be slight variations in the order in which different channels get them it's unlikely to be a noticeable difference.